### PR TITLE
State machine inside transport

### PR DESCRIPTION
**Pull request by olivakar**
Now the transport is no longer a state machine by itself.
As before, the `MQTTTransport` extends `AbstractTransport`

The method names for the triggers have been changed to `trig_method()` so that they are not simlar to the methods that are end points.

I was able to make the transport a model of the state machine as discussed between me and @BertKleewein

### CODE REVIEW

File: azure-iot-hub-devicesdk/azure/iot/hub/devicesdk/internal_client.py [hunk 1]
```diff
@@ -37,6 +37,7 @@ def connect(self):
     def send_event(self, event):
         """
         Sends an actual message/telemetry to the IoT Hub via the message broker.
+        This is an outgoing events and traditionally means true "telemetry".
```
[the remark this answers is not in the document]
Comment (pierreca) [reply azure-iot-hub-devicesdk/azure/iot/hub/devicesdk/internal_client.py]: I don't think that's great docs. requirement files are internal cuisine, we must do better for public docs. 
Suggestion:
"Sends a message to the default events endpoint on the Azure IoT Hub or Edge Hub instance"

---
In reply to: [234891767](https://github.com/Azure/azure-iot-sdk-python-preview/pull/22#discussion_r234891767) [](ancestors = 234891767,234843865)

File: azure-iot-hub-devicesdk/azure/iot/hub/devicesdk/transport/mqtt/mqtt_transport.py [hunk 2]
```diff
@@ -3,101 +3,100 @@
...
             },
             {
-                "trigger": "connect",
+                "trigger": "trig_connect",
                 "source": ["connecting", "connected", "sending"],
                 "dest": None,
             },
             {
-                "trigger": "_provider_connect_complete",
+                "trigger": "trig_provider_connect_complete",
```
[the remark this answers is not in the document]
Comment (pierreca) [reply azure-iot-hub-devicesdk/azure/iot/hub/devicesdk/transport/mqtt/mqtt_transport.py]: that's not what I'm asking though. I don't really care whether there's an underscore or not - but I need the team to work using the same conventions. Is there a convention? what is it? why was it removed?

---
In reply to: [234892532](https://github.com/Azure/azure-iot-sdk-python-preview/pull/22#discussion_r234892532) [](ancestors = 234892532,234844365)

File: azure-iot-hub-devicesdk/azure/iot/hub/devicesdk/transport/mqtt/mqtt_transport.py [hunk 3]
```diff
@@ -181,7 +189,9 @@ def _after_action_deliver_next_queued_event(self, event):
             topic = self._get_telemetry_topic()
             self._mqtt_provider.publish(topic, event_to_send)
         except queue.Empty:
-            logger.warning("UNEXPECTED: queue is empty in _after_action_deliver_next_queued_event")
+            logger.warning(
+                "UNEXPECTED: queue is empty in _after_action_deliver_next_queued_event"
+            )
```
[the remark this answers is not in the document]
Comment (pierreca) [reply azure-iot-hub-devicesdk/azure/iot/hub/devicesdk/transport/mqtt/mqtt_transport.py]: please make sure you run black with the line-length set to 100

---
In reply to: [234892033](https://github.com/Azure/azure-iot-sdk-python-preview/pull/22#discussion_r234892033) [](ancestors = 234892033,234846394)

File: azure-iot-hub-devicesdk/azure/iot/hub/devicesdk/transport/mqtt/mqtt_transport.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (olivakar) [reply azure-iot-hub-devicesdk/azure/iot/hub/devicesdk/transport/mqtt/mqtt_transport.py]: I removed it as i felt it was sufficient to start the function with the word `trig`  which would accomplish 2 things :-These are triggers and These are internal on the state machine and should not be called from outside. But I think for making it more explicit that these are not public we should start the name with `_trig`

---
In reply to: [235141244](https://github.com/Azure/azure-iot-sdk-python-preview/pull/22#discussion_r235141244) [](ancestors = 235141244,234892532,234844365)

**Comment by BertKleewein** [comment]
For what it's worth, I added the underscores when the trigger functions were added as part of the transport object as a way to say "consumers of the transport should not call this function."

I can't remember if this change moves the triggers into the machine member or if they're still on the transport.  (I'm on my phone).  If they're still on the transport, I think they should all be underscored.  If they're on the machine, then I think it might not matter.

**Comment by BertKleewein** [comment]
Is there a script that calls black or a config file that we can add?  Remembering to add command line args every time is not my strong suit.

**Comment by olivakar** [comment]
There is a way to add black to `Pycharm` and you can run it from the IDE. I am using it in this manner. We can also run `black` every time we save a file in `Pycharm` though I have not configured that for `Pycharm` yet.

**Pull request merged**